crates/e2e/src/drink_client.rs

```diff
@@ -121,12 +127,39 @@ impl<AccountId: AsRef<[u8; 32]> + Send, Hash> ChainBackend for Client<AccountId,
 
     async fn runtime_call<'a>(
         &mut self,
-        _origin: &Keypair,
-        _pallet_name: &'a str,
-        _call_name: &'a str,
-        _call_data: Vec<Value>,
+        origin: &Keypair,
+        pallet_name: &'a str,
+        call_name: &'a str,
+        call_data: Vec<Value>,
     ) -> Result<Self::EventLog, Self::Error> {
-        todo!("https://github.com/Cardinal-Cryptography/drink/issues/36")
+        // Since in general, `ChainBackend::runtime_call` must be dynamic, we have to
+        // perform some translation here in order to invoke strongly-typed drink!
+        // API.
+
+        // Get metadata of the drink! runtime, so that we can encode the call object.
+        // Panic on error - metadata of the static im-memory runtime should always be
+        // available.
+        let raw_metadata: Vec<u8> = MinimalRuntime::metadata().into();
```

Comment:
What if I want to write an e2e test for a runtime that is different than `MinimalRuntime`? For example Aleph Zero liminal one? I won't be able to use the drink! as backend?

Reply:
I will add this feature soon

Reply:
@deuszx https://github.com/paritytech/ink/pull/1892